plugin/src/main/java/org/opensearch/ml/task/MLTaskManager.java

```diff
@@ -541,32 +541,81 @@ private ActionListener<UpdateResponse> getUpdateResponseListener(String taskId,
         });
     }
 
-    public void startTaskPollingJob() throws IOException {
+    public void startTaskPollingJob() {
         if (this.taskPollingJobStarted) {
             return;
         }
 
-        String id = "ml_batch_task_polling_job";
-        String jobName = "poll_batch_jobs";
-        String interval = "1";
-        Long lockDurationSeconds = 20L;
-
-        MLJobParameter jobParameter = new MLJobParameter(
-            jobName,
-            new IntervalSchedule(Instant.now(), Integer.parseInt(interval), ChronoUnit.MINUTES),
-            lockDurationSeconds,
-            null,
-            MLJobType.BATCH_TASK_UPDATE
-        );
-        IndexRequest indexRequest = new IndexRequest()
-            .index(CommonValue.ML_JOBS_INDEX)
-            .id(id)
-            .source(jobParameter.toXContent(JsonXContent.contentBuilder(), null))
-            .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
-
-        client.index(indexRequest, ActionListener.wrap(r -> {
-            log.info("Indexed ml task polling job successfully");
-            this.taskPollingJobStarted = true;
-        }, e -> log.error("Failed to index task polling job", e)));
+        mlIndicesHandler.initMLJobsIndex(ActionListener.wrap(success -> {
+            if (success) {
+                String id = "ml_batch_task_polling_job";
+                String jobName = "poll_batch_jobs";
+                String interval = "1";
+                Long lockDurationSeconds = 20L;
+
+                MLJobParameter jobParameter = new MLJobParameter(
+                    jobName,
+                    new IntervalSchedule(Instant.now(), Integer.parseInt(interval), ChronoUnit.MINUTES),
+                    lockDurationSeconds,
+                    null,
+                    MLJobType.BATCH_TASK_UPDATE
+                );
+                IndexRequest indexRequest = new IndexRequest()
+                    .index(CommonValue.ML_JOBS_INDEX)
+                    .id(id)
+                    .source(jobParameter.toXContent(JsonXContent.contentBuilder(), null))
+                    .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
+
+                try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+                    client.index(indexRequest, ActionListener.runBefore(ActionListener.wrap(r -> {
+                        log.info("Indexed ml task polling job successfully");
+                        this.taskPollingJobStarted = true;
+                    }, e -> log.error("Failed to index task polling job", e)), context::restore));
```

Comment:
Both methods have similar patterns for creating and indexing jobs. Consider extracting common functionality into a reusable method.
private void indexMLJob(IndexRequest indexRequest, MLJobType jobType) {
    try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
        client.index(indexRequest, ActionListener.runBefore(ActionListener.wrap(
            r -> log.info("Indexed {} job successfully", jobType),
            e -> log.error("Failed to index " + jobType + " job", e)
        ), context::restore));
    }
}